src/gennylib/test/conventions_test.cpp

```diff
@@ -1,37 +1,112 @@
 #include "test.h"
 
 #include <chrono>
+#include <cmath>
 
 #include <yaml-cpp/yaml.h>
 
 #include <gennylib/conventions.hpp>
 
+namespace genny {
+namespace {
 using namespace std::chrono;
 
-TEST_CASE("Millisecond conversions") {
-    SECTION("Can convert to milliseconds") {
-        REQUIRE(YAML::Load("D: 300")["D"].as<milliseconds>().count() == 300);
-        REQUIRE(YAML::Load("-1").as<milliseconds>().count() == -1);
-        REQUIRE(YAML::Load("0").as<milliseconds>().count() == 0);
+TEST_CASE("genny::TimeSpec conversions") {
+    SECTION("Can convert to genny::TimeSpec") {
+        REQUIRE(YAML::Load("D: 3 seconds")["D"].as<TimeSpec>().count() == 3 * std::pow(10, 9));
+        REQUIRE(YAML::Load("0 second").as<TimeSpec>().count() == 0);
+        REQUIRE(YAML::Load("20 millisecond").as<TimeSpec>().count() == 20 * std::pow(10, 6));
+        REQUIRE(YAML::Load("33 microsecond").as<TimeSpec>().count() == 33 * std::pow(10, 3));
+        REQUIRE(YAML::Load("2e3 microseconds").as<TimeSpec>().count() == 2000 * std::pow(10, 3));
+        REQUIRE(YAML::Load("10.3e2 nanoseconds").as<TimeSpec>().count() == 1030);
+        REQUIRE(YAML::Load("3 hour").as<TimeSpec>().count() == 3 * 3600 * std::pow(10, 9));
+        REQUIRE(YAML::Load("2 minutes").as<TimeSpec>().count() == 2 * 60 * std::pow(10, 9));
+    }
+
+    SECTION("Overlooks small typos") {
+        REQUIRE(YAML::Load("D: 3 secondsasdfadsf     ")["D"].as<TimeSpec>().count() ==
+                3 * std::pow(10, 9));
     }
 
     SECTION("Barfs on unknown types") {
-        REQUIRE_THROWS(YAML::Load("foo").as<milliseconds>());
-        REQUIRE_THROWS(YAML::Load("[1,2,3]").as<milliseconds>());
-        REQUIRE_THROWS(YAML::Load("[]").as<milliseconds>());
-        REQUIRE_THROWS(YAML::Load("{}").as<milliseconds>());
-        REQUIRE_THROWS(YAML::Load("foo: 1").as<milliseconds>());
+        REQUIRE_THROWS(YAML::Load("-1 nanosecond").as<TimeSpec>());
+        REQUIRE_THROWS(YAML::Load("foo").as<TimeSpec>());
+        REQUIRE_THROWS(YAML::Load("[1,2,3]").as<TimeSpec>());
+        REQUIRE_THROWS(YAML::Load("[]").as<TimeSpec>());
+        REQUIRE_THROWS(YAML::Load("{}").as<TimeSpec>());
+        REQUIRE_THROWS(YAML::Load("what nanoseconds").as<TimeSpec>());
+        REQUIRE_THROWS(YAML::Load("29 picoseconds").as<TimeSpec>());
+        REQUIRE_THROWS(YAML::Load("1e3 centuries").as<TimeSpec>());
+        REQUIRE_THROWS(YAML::Load("mongodb").as<TimeSpec>());
+        REQUIRE_THROWS(YAML::Load("1").as<TimeSpec>());
+        REQUIRE_THROWS(YAML::Load("333").as<TimeSpec>());
```

Comment:
Waiiiiiiit, does this mean that just a number is no longer pure millisecond? This is the sort of config schema break that caused trouble around NoOp. Maybe the mailing list should be given a poke about this.

Reply:
Ah good point. I will send an email out today.